conans/test/functional/options/options_test.py

```diff
@@ -205,32 +193,34 @@ def configure(self):
     def general_scope_options_test(self):
         # https://github.com/conan-io/conan/issues/2538
         client = TestClient()
-        conanfile_libA = """from conans import ConanFile
-class LibA(ConanFile):
-    options = {"shared": [True, False]}
+        conanfile_liba = textwrap.dedent("""
+            from conans import ConanFile
+            class LibA(ConanFile):
+                options = {"shared": [True, False]}
 
-    def configure(self):
-        self.output.info("shared=%s" % self.options.shared)
-    """
-        client.save({"conanfile.py": conanfile_libA})
+                def configure(self):
+                    self.output.info("shared=%s" % self.options.shared)
+                """)
+        client.save({"conanfile.py": conanfile_liba})
         client.run("create . libA/0.1@danimtb/testing -o *:shared=True")
         self.assertIn("libA/0.1@danimtb/testing: shared=True", client.out)
 
-        conanfile_libB = """from conans import ConanFile
-class LibB(ConanFile):
-    options = {"shared": [True, False]}
-    requires = "libA/0.1@danimtb/testing"
+        conanfile_libb = textwrap.dedent("""
+            from conans import ConanFile
+            class LibB(ConanFile):
+                options = {"shared": [True, False]}
+                requires = "libA/0.1@danimtb/testing"
 
-    def configure(self):
-        self.options["*"].shared = self.options.shared
-        self.output.info("shared=%s" % self.options.shared)
-    """
+                def configure(self):
+                    self.options["*"].shared = self.options.shared
+                    self.output.info("shared=%s" % self.options.shared)
+                """)
 
         for without_configure_line in [True, False]:
             if without_configure_line:
-                conanfile = conanfile_libB.replace(
-                    "        self.options[\"*\"].shared = self.options.shared", "")
-
+                conanfile = conanfile_libb.replace("self.options[", "#")
```

Comment:
This was wrongly testing twice the same thing.

Reply:
Yes, without changing anything else everything keeps working... I wonder what was the purpose of the line:

```
self.options["*"].shared = self.options.shared
```

It looks like a call like `conan create . libB/0.1@danimtb/testing -o libB:shared=True` is missing... to check that with and without that line the assignment for `libA` is different. Or maybe `conan create . libB/0.1@danimtb/testing -o libB:shared=True -o *:shared=False`
```